app/controllers/links_controller.rb

```diff
@@ -1,38 +1,54 @@
 class LinksController < ApplicationController
-  include Concerns::Utility
-  before_action :confirm_logged_in, except: [:index, :create]
-  before_action :normalize_params, only: [:update]
-  before_action :find_link, only: [
-                                    :show,
-                                    :update,
-                                    :destroy,
-                                    :toggle_activate
-                                  ]
+  before_action :confirm_logged_in,
+                except: [:index, :create, :redirect, :inactive, :deleted]
+  before_action :find_link,
+                except: [:index, :create, :show, :redirect, :inactive, :deleted]
+  layout 'dashboard', only: [:show]
 
   def index
     @index_presenter = Index::IndexPresenter.new
     @new_link = Link.new
   end
 
   def create
-    @link = Link.new(normalize_params)
-    update_current_user(@link)
+    @link = Link.new(link_params)
+    current_user.add_new_link(@link) if current_user
     if @link.save
-      respond_to_save
+      set_flash_and_redirect
     else
       flash[:error] = new_link_error
       redirect_to :back
     end
   end
 
+  def show
+    @links = Link.find_links_for_user(current_user)
+    @link = Link.find_by_id(params[:link_id])
```

Comment:
It's recommended that you use the `find_by` method like so
 `find_by(id: params[:link_id])`